poetry.lock

```diff
@@ -1007,6 +1007,21 @@ python-versions = ">=3.7"
 [package.extras]
 idna2008 = ["idna"]
 
+[[package]]
+name = "rust-python-jaeger-reporter"
+version = "0.1.20"
+description = "A faster reporter for the python `jaeger-client` that reports spans in a native background thread."
+category = "main"
+optional = false
+python-versions = ">=3.6"
+develop = false
+
+[package.source]
+type = "git"
+url = "https://github.com/erikjohnston/rust-jaeger-python-client.git"
+reference = "master"
+resolved_reference = "27754b5f56b4099374c389f677911f7af852a00e"
```

Comment:
With the `rust-python-jaeger-reporter` install from repo, seeing this error: `ERROR: Can't verify hashes for these requirements because we don't have a way to hash version control repositories:`

```
#20 [builder 6/9] RUN --mount=type=cache,target=/root/.cache/pip   pip install --prefix="/install" --no-deps --no-warn-script-location -r /synapse/requirements.txt
#20 sha256:f468e89a00edbfafedfe3187c7ff953d24eada40bf087dd90a2f3758266a3416
#20 4.037 Ignoring importlib-metadata: markers 'python_full_version >= "3.7.1" and python_version < "3.8"' don't match your environment
#20 4.037 Ignoring importlib-resources: markers 'python_full_version >= "3.7.1" and python_version < "3.9"' don't match your environment
#20 4.041 Ignoring psycopg2cffi-compat: markers 'python_full_version >= "3.7.1" and python_full_version < "4.0.0" and platform_python_implementation == "PyPy"' don't match your environment
#20 4.042 Ignoring psycopg2cffi: markers 'python_full_version >= "3.7.1" and python_full_version < "4.0.0" and platform_python_implementation == "PyPy"' don't match your environment
#20 4.047 Collecting rust-python-jaeger-reporter@ git+https://github.com/erikjohnston/rust-jaeger-python-client.git@master
#20 4.049 ERROR: Can't verify hashes for these requirements because we don't have a way to hash version control repositories:
#20 4.049     rust-python-jaeger-reporter@ git+https://github.com/erikjohnston/rust-jaeger-python-client.git@master from git+https://github.com/erikjohnston/rust-jaeger-python-client.git@master (from -r /synapse/requirements.txt (line 610))
```

Reply:
Looks like we can use `TEST_ONLY_SKIP_DEP_HASH_VERIFICATION`, https://github.com/matrix-org/synapse/blob/bf3115584c876392e7849cd4ba2271165572588d/docker/Dockerfile#L66-L69

Reply:
Putting it together, tests are passing:

```
TEST_ONLY_SKIP_DEP_HASH_VERIFICATION=1 COMPLEMENT_ALWAYS_PRINT_SERVER_LOGS=1 COMPLEMENT_DIR=../complement ./scripts-dev/complement.sh -run TestMessagesOverFederation -p 1
[...]
synapse_main | 2022-07-26 22:13:14,275 - synapse.logging.opentracing - 439 - INFO - sentinel - Using rust_python_jaeger_reporter library
[...]

--- PASS: TestMessagesOverFederation (107.87s)
    --- PASS: TestMessagesOverFederation/parallel (0.00s)
        --- PASS: TestMessagesOverFederation/parallel/asdf (34.27s)
PASS
ok  	github.com/matrix-org/complement/tests	113.237s
2022/07/26 17:14:23 config: &{BaseImageURI:complement-synapse BaseImageArgs:[] DebugLoggingEnabled:false AlwaysPrintServerLogs:true BestEffort:false EnvVarsPropagatePrefix:PASS_ SpawnHSTimeout:30s KeepBlueprints:[] HostMounts:[] PackageNamespace:csapi CACertificate:0xc000bd6000 CAPrivateKey:0xc0000c1ec0}
testing: warning: no tests to run
PASS
ok  	github.com/matrix-org/complement/tests/csapi	1.536s [no tests to run]
```

But with `rust-python-jaeger-reporter`, I'm not seeing any traces make their way back to Jaeger though yet.

Reply:
> But with `rust-python-jaeger-reporter`, I'm not seeing any traces make their way back to Jaeger though yet.

Probably because `rust-python-jaeger-reporter` ["is not configurable and is hardcoded to report to the local agent on localhost and the default port."](https://github.com/erikjohnston/rust-jaeger-python-client/tree/27754b5f56b4099374c389f677911f7af852a00e#limitations). Probably have to update to pull from `jaeger_config.local_agent.reporting_host` like the normal client does